src/test/java/tennisclub/dao/RankingDaoImplTest.java

```diff
@@ -0,0 +1,215 @@
+import tennisclub.entity.User;
+import tennisclub.entity.ranking.Ranking;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.transaction.Transactional;
+import javax.validation.ConstraintViolationException;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@Transactional
+@SpringBootTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+class RankingDaoImplTest {
+    private Tournament tournament;
+    private Tournament otherTournament;
+    private User user;
+    private User otherUser;
+
+    @Autowired
+    RankingDao rankingDao;
+
+    @PersistenceContext
+    EntityManager manager;
+
+    @BeforeEach
+    void initRankingDeps() {
+        Court court = new Court();
+        court.setName("My Court");
+        manager.persist(court);
+
+        tournament = new Tournament();
+        tournament.setStartTime(LocalDateTime.now());
+        tournament.setEndTime(LocalDateTime.now().plusHours(1));
+        tournament.setPrize(10000);
+        tournament.setCourt(court);
+        manager.persist(tournament);
+
+        otherTournament = new Tournament();
+        otherTournament.setStartTime(LocalDateTime.now().plusHours(1));
+        otherTournament.setEndTime(LocalDateTime.now().plusHours(2));
+        otherTournament.setPrize(5000);
+        otherTournament.setCourt(court);
+        manager.persist(otherTournament);
+
+        user = new User();
+        user.setUsername("emil62");
+        manager.persist(user);
+
+        otherUser = new User();
+        otherUser.setUsername("winston84");
+        manager.persist(otherUser);
+    }
+
+    @Test
+    void nullTournament() {
+        assertThrows(
+                ConstraintViolationException.class,
+                () -> {
+                    rankingDao.create(new Ranking(null, user));
+                    manager.flush();
+                }
+        );
+    }
+
+    @Test
+    void nullUser() {
+        assertThrows(
+                ConstraintViolationException.class,
+                () -> {
+                    rankingDao.create(new Ranking(tournament, null));
+                    manager.flush();
+                }
+        );
+    }
+
+    @Test
+    void primaryKeyUniqueness() {
+        assertThrows(
+                DataIntegrityViolationException.class,
+                () -> {
+                    rankingDao.create(new Ranking(tournament, user));
+                    rankingDao.create(new Ranking(tournament, user));
+                }
+        );
+    }
+
+    @Test
+    void create() {
+        Ranking createdRanking = new Ranking(tournament, user);
+        createdRanking.setPlayerPlacement(42);
+        rankingDao.create(createdRanking);
+
+        Ranking ranking = manager.createQuery("select r from Ranking r", Ranking.class).getSingleResult();
```

Comment:
Can't this return a different result if the DB has more than one ranking persisted? 👀